sql/src/main/java/io/crate/operation/collect/files/SummitsIterable.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.collect.files;
+
+import com.google.common.base.Splitter;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.common.primitives.Ints;
+import io.crate.types.DataTypes;
+import org.elasticsearch.node.internal.InternalSettingsPreparer;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class SummitsIterable implements Supplier<Iterable<?>> {
+	
+	private static final Splitter TAB_SPLITTER = Splitter.on("\t");
+
+	@Override
+	public Iterable<?> get() {
+		return summitsSupplierCache.get();
+	}
+
+	private final Supplier<Set<SummitsContext>> summitsSupplierCache = Suppliers.memoizeWithExpiration(
```

Comment:
@matriv so most of the time the table won't be used, so maybe reading it once and keeping it always in memory is not that space effective. Therefore, it would be probably better to lazy load the summits set and have the expiration on it.